vre/values.yaml

```diff
@@ -0,0 +1,485 @@
+    config:
+      JupyterHub:
+        authenticator_class: "generic-oauth"
+      RucioAuthenticator:
+        # client_id: "" # set through secret
+        # client_secret: "" #  set through secret
+        authorize_url: https://iam-escape.cloud.cnaf.infn.it/authorize
+        token_url: https://iam-escape.cloud.cnaf.infn.it/token
+        userdata_url: https://iam-escape.cloud.cnaf.infn.it/userinfo
+        oauth_callback_url: https://jhub-vre.obsuks4.unige.ch/hub/oauth_callback
+        username_key: preferred_username
+        enable_auth_state: true
+        allow_all: true        
+        scope:
+          - openid
+          - profile
+          - email
+
+    
+    extraConfig:
+
+      token-exchange: |
+        import pprint
+        import os
+        import warnings
+        import requests
+        from oauthenticator.generic import GenericOAuthenticator
+
+        # custom authenticator to enable auth_state and get access token to set as env var for rucio extension
+        class RucioAuthenticator(GenericOAuthenticator):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+                self.enable_auth_state = True
+
+            def exchange_token(self, token):
+                params = {
+                    'client_id': self.client_id,
+                    'client_secret': self.client_secret,
+                    'grant_type': 'urn:ietf:params:oauth:grant-type:token-exchange',
+                    'subject_token_type': 'urn:ietf:params:oauth:token-type:access_token',
+                    'subject_token': token,
+                    'scope': 'openid profile',
+                    'audience': 'rucio'
+                }
+                response = requests.post(self.token_url, data=params)
+                print("EXCHANGE TOKEN for params", params)
+                print(response.json())
+                rucio_token = response.json()['access_token']
+
+                return rucio_token
+                
+            async def pre_spawn_start(self, user, spawner):
+                auth_state = await user.get_auth_state()
+                #print("AUTH_state")
+                #pprint.pprint(auth_state)
+                if not auth_state:
+                    # user has no auth state
+                    return False
+                
+                # define token environment variable from auth_state
+                spawner.environment['RUCIO_ACCESS_TOKEN'] = self.exchange_token(auth_state['access_token'])
+                spawner.environment['EOS_ACCESS_TOKEN'] = auth_state['access_token']
+        
+        # set the above authenticator as the default
+        c.JupyterHub.authenticator_class = RucioAuthenticator
+
+        # enable authentication state
+        c.GenericOAuthenticator.enable_auth_state = True
+      
+  singleuser:
+    # TODO: revert!
+    cloudMetadata:
+      blockWithIptables: false 
+
+    extraEnv:        
+      RUCIO_AUTH_HOST: "https://vre-rucio-auth.cern.ch"
```

Comment:
I think this should move to a top-level configuration as eventually I think we will want to configure REANA to use rucio (with token auth)